### DISCUSSION TITLE
mail rule re-executed after manually moving mail back to inbox

### DISCUSSION THREAD

**Discussion by ielfretter** [tutao/tutanota#5736] [Ideas]
Hi,

If you try the following scenario:

- Create a mail rule
- Test the mail rule and note that it is executed as intended
- Manually move the mail back to the inbox
- Wait for any other mail to arrive
-  **OR** restart the desktop client or iOS app

Then the mail-rule will be re-executed. Not sure if this is by design, but personally I think that once the mail has been manually moved back to the mailbox it should not be re-executed. A rule - to my opinion - should only be triggered once, upon first time reception.

```
--
Client: Mac Desktop
Tutanota version: 3.115.2
Time zone: Europe/Amsterdam
User agent:
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)

**Comment by vitoreiji** [comment]
IIRC, inbox rules are applied to unread messages in your inbox. Your suggestion does make sense, though, so I'll consider it a feature request.

**Reply by charlag** [reply]
I remember @bedhub doing some tricks with re-sorting email by moving it to inbox

**Reply by vitoreiji** [reply]
Yeah, that's a cool trick. If we implement this we should also allow triggering inbox rules manually on any folder.
